### PR TITLE
Put

### PR DESCRIPTION
**Pull request by westerbarendregt**
Build an language interpretator and the chunk data is now working, tester now stuck on a 405 (bad status code) what we have as default in POST

### CODE REVIEW

File: srcs/RequestHandler.cpp [hunk 1]
```diff
@@ -524,8 +537,8 @@ std::string		RequestHandler::handlePUT()
 	else 
 		this->m_request_data->m_status_code = 201;
 	char* current_dir = getcwd(NULL, 0);
-	
-	if (chdir(upload_store))
+	upload_store = upload_store.substr(0, upload_store.size() - 1); // getting an extra space.
```
Comment (westerbarendregt) [inline srcs/RequestHandler.cpp]: So this was giving me quite some brain farts, it seems that it's the same problem the other parser was also giving. [An extra space at the end]. This works now, but maybe we want to check the initial getter.

File: srcs/Authentication.cpp [hunk 2]
```diff
@@ -60,3 +60,24 @@ void                Authenticated(Client& c, RequestHandler& req)
     CheckCorrectCredentials(decoded, path_ht);
 	Logger::Log() << "[USER-AGENT = AUTHENTICATED]" << std::endl;
 }
+
+void                GetLanguage(Client& c, RequestHandler& req)
+{
+    std::string content_language = c.getRequest().m_headers[CONTENTLANGUAGE];
+    if (content_language[0] == 0){
```
Comment (borispeeters) [inline srcs/Authentication.cpp]: maybe change this to:
if (content_language.empty()) {

File: www/language/index.html.de [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Willkommen bei webserv, jaah!</title>
```
Comment (borispeeters) [inline www/language/index.html.de:4]: lol

File: srcs/Authentication.cpp (hunk 2 above)
Comment (westerbarendregt) [reply srcs/Authentication.cpp]: Yes sir

File: srcs/RequestHandler.cpp [hunk 4]
```diff
@@ -513,9 +526,11 @@ void	RequestHandler::handleMetadata(t_client &c) {
 std::string		RequestHandler::handlePUT()
 {
 	std::string m_file = this->m_request_data->m_real_path.substr(this->m_request_data->m_real_path.find_last_of('/') + 1);
-	const char *upload_store = this->m_request_data->m_location->second["upload_store"].c_str();
-	std::string path_to_file = std::string(upload_store) + m_file;
+	std::string upload_store = this->m_request_data->m_location->second["upload_store"];
+	upload_store.erase(upload_store.find_last_of(' '));
+	std::string path_to_file = std::string(upload_store) + '/' + m_file;
 
+	std::cout << "path: " << path_to_file << std::endl;
```
Comment (borispeeters) [inline srcs/RequestHandler.cpp]: change to Logger::Log()

**Review by borispeeters — CHANGES_REQUESTED** [review]
Something I found: if you give for example:
Content-Language: nlnothappeningmate
as header, it will recognize it as dutch, even though it should probably reject it

File: srcs/RequestHandler.cpp (hunk 4 above)
Comment (westerbarendregt) [reply srcs/RequestHandler.cpp]: yes, took it out

**Comment by westerbarendregt** [comment]
couldn't see how to reply to your last comment, but It should be fixed now. in this case it won't work:
nlblabla or nl-blabla
but it will work with: 
nl-bl and nl, blabla
The content-language would work for example with en-CA, which would mean English-Canadian. I don't suppose we would have to check which languages exist and which don't.

**Comment by borispeeters** [comment]
yeah it seems to work
an alternative idea I had is maybe we could make an std::vector or map of supported languages
Also what you implemented is looking up a file based on a request headers, shouldn't that be accept-language instead of content-language? and then in the response we send back the content-language header

**Pull request closed without merging**